From 87e687923c02b9fb554005695d74679d045df61c Mon Sep 17 00:00:00 2001
From: Adam Duskett <adam.duskett@amarulasolutions.com>
Date: Tue, 6 Feb 2024 20:14:19 +0100
Subject: [PATCH] fix boost-log dynamic linking

If this definition does not exist, linking fails with hundreds of errors about
unresolved definitions in boost::log.

Upstream: Not submitted

Signed-off-by: Adam Duskett <adam.duskett@amarulasolutions.com>
---
 cmake/boost.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cmake/boost.cmake b/cmake/boost.cmake
index dfa02f2..3ea7264 100644
--- a/cmake/boost.cmake
+++ b/cmake/boost.cmake
@@ -1,6 +1,7 @@
 # Code to import Boost correctly, either from the system, or from a package.
 
 find_package(Boost 1.74 COMPONENTS log)
+ADD_DEFINITIONS(-DBOOST_LOG_DYN_LINK)
 
 option(MENDER_DOWNLOAD_BOOST "Download Boost if it is not found (Default: OFF)" OFF)
 
-- 
2.43.0

